src/hooks/useDelegateAndUndelegateFunds.ts

```diff
@@ -0,0 +1,127 @@
+import { toaster } from 'toasterhea'
+import DelegateFundsModal from '~/modals/DelegateFundsModal'
+import { Layer } from '~/utils/Layer'
+import { BN, toBN } from '~/utils/bn'
+import {
+    delegateToOperator,
+    getOperatorDelegationAmount,
+    undelegateFromOperator,
+} from '~/services/operators'
+import getChainId from '~/utils/web3/getChainId'
+import { getConfigForChain } from '~/shared/web3/config'
+import { getCustomTokenBalance } from '~/marketplace/utils/web3'
+import { OperatorElement } from '~/types/operator'
+import { useConfigFromChain } from '~/hooks/useConfigFromChain'
+import { useWalletAccount } from '~/shared/stores/wallet'
+import { waitForGraphSync } from '~/getters/waitForGraphSync'
+import UndelegateFundsModal from '~/modals/UndelegateFundsModal'
+import getSponsorshipTokenInfo from '~/getters/getSponsorshipTokenInfo'
+import { fromDecimals } from '~/marketplace/utils/math'
+export const useDelegateAndUndelegateFunds = (): {
+    delegateFunds: (operator: OperatorElement) => Promise<void>
+    undelegateFunds: (operator: OperatorElement) => Promise<void>
+} => {
+    const delegateFundsModal = toaster(DelegateFundsModal, Layer.Modal)
+    const undelegateFundsModal = toaster(UndelegateFundsModal, Layer.Modal)
+    const walletAddress = useWalletAccount()
+    const { minimumSelfDelegationFraction } = useConfigFromChain()
+
+    const getDelegationAmount = async (
+        operator: OperatorElement,
+        address: string,
+    ): Promise<BN> => {
+        return await getOperatorDelegationAmount(operator.id, address)
+    }
```

Comment:
This can also go outside. Id does not depend on anything reacty.

Reply:
true, I removed that and used direct calls to `getOperatorDelegationAmount`